pkg/cli/webhook.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+
+	"github.com/spf13/cobra"
+)
+
+func (c CLI) newCreateWebhookCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "webhook",
+		Short: "Scaffold a webhook for an API resource",
+		Long: `Scaffold a webhook for an API resource.
+`,
+		RunE: errCmdFunc(
+			fmt.Errorf("webhook subcommand requires an existing project"),
+		),
+	}
+
+	found, projectVersion := getProjectVersion()
+	if !found {
+		msg := `For project-specific information, run this command in the root directory of a
+project.
+`
+		cmd.Long = fmt.Sprintf("%s\n%s", cmd.Long, msg)
+		return cmd
+	}
+
+	// Lookup the plugin for projectVersion and bind it to the command.
+	c.bindCreateWebhook(cmd, projectVersion)
+	return cmd
+}
+
+func (c CLI) bindCreateWebhook(cmd *cobra.Command, projectVersion string) {
+	ps, ok := c.plugins[projectVersion]
+	if !ok {
+		err := fmt.Errorf("unknown project version %q", projectVersion)
+		cmdErr(cmd, err)
+		return
+	}
+	var cwg plugin.CreateWebhookPluginGetter
+	var hasGetter bool
+	for _, p := range ps {
+		if hasGetter {
+			err := fmt.Errorf("duplicate webhook creation plugins for project version %q: %s, %s",
+				projectVersion, cwg.Name(), p.Name())
+			cmdErr(cmd, err)
+			return
+		}
+		cwg, hasGetter = p.(plugin.CreateWebhookPluginGetter)
+	}
+	if !hasGetter {
+		err := fmt.Errorf("project version %q does not support a webhook creation plugin",
+			projectVersion)
+		cmdErr(cmd, err)
+		return
+	}
+
+	cwp := cwg.GetCreateWebhookPlugin()
```

Comment:
This logic seems like we can add a helper since it's pretty identical across commands.

Reply:
Will refactor in a follow-up if possible.